### PR TITLE
Port design_docs.js tests from js into elixir

### PR DESCRIPTION
**Pull request by jjrodrig**
## Overview
This PR ports the design_docs.js test suite into the elixir test suite

## Testing recommendations
```
make elixir tests=test/elixir/test/design_docs_test.exs
```

## Related Issues or Pull Requests
N/A

## Checklist

- [X] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: test/javascript/tests/design_docs.js [hunk 1]
```diff
@@ -10,7 +10,9 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
+couchTests.elixir = true;
```
Comment (dottorblaster) [inline test/javascript/tests/design_docs.js]: Is this something new?

Comment (jjrodrig) [reply test/javascript/tests/design_docs.js]: I just noticed that this is the correct way to tag the porting progress. I took a wrong reference for this.

Comment (dottorblaster) [reply test/javascript/tests/design_docs.js]: Oh! So we'll have to upgrade some :) the rest is perfect, just merge 👍

Comment (jjrodrig) [reply test/javascript/tests/design_docs.js]: Yes I think we should check the ported tests and tag it correctly

**Pull request merged**